### DISCUSSION TITLE
GraphQL Yoga subcriptions over WebSocket

### DISCUSSION THREAD

**Discussion by lewebsimple** [nitrojs/nitro#2938] [Q&A]
Hey everyone :wave:

I'd like to configure GraphQL subscriptions using GraphQL Yoga in a Nuxt app with the experimental websocket feature of Nitro, but I don't know for sure where to start from.

I had successfully done this by initializing the graphql-ws server inside a server middleware but it does seem hackish and only works in Node environments.

Any insight would be greatly appreciated!

**Comment by memotux** [comment]
Hi  👋.

I have use `graphql-ws` as a Nitro plugin. If you want to see my implementation you can see the code in my repository: https://github.com/memotux/portfolio-graphql-ws/blob/master/nitro/plugins/ws.ts

I see that my implementation could be adapted as Graphql Yoga WS implementation: https://the-guild.dev/graphql/yoga-server/docs/features/subscriptions#graphql-over-websocket-protocol-via-graphql-ws

Hope this help.
